test/e2e/addon/metricsserver.go

```diff
@@ -0,0 +1,104 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	metricsv1beta1 "k8s.io/metrics/pkg/client/clientset/versioned"
+)
+
+const (
+	metricsServerNamespace = "kube-system"
+	metricsServerName      = "metrics-server"
+)
+
+type MetricsServerAddon struct {
+	Addon
+	cfg *rest.Config
+}
+
+func NewMetricsServerAddon(cluster string, cfg *rest.Config) AddonIface {
+	return MetricsServerAddon{
+		Addon: Addon{
+			Cluster:   cluster,
+			Name:      metricsServerName,
+			Namespace: metricsServerNamespace,
+		},
+		cfg: cfg,
+	}
+}
+
+func (m MetricsServerAddon) Setup(ctx context.Context, eksClient *eks.Client, k8s kubernetes.Interface, logger logr.Logger) error {
+	return nil
+}
+
+func (m MetricsServerAddon) PostInstall(ctx context.Context, eksClient *eks.Client, k8s kubernetes.Interface, logger logr.Logger) error {
+	return nil
+}
```

Comment:
good indicator that you are defining the interface too early
You already have methods without a usecase